Q:

ROS Answers SE migration: map building

I am new in ROS and self studying about this. Now I am working on map building using turtlebot simulation. I came across map building and navigation. When I tried (page:7):
$ roslaunch ros_liv_turtlebot office_turtlebot.launch
it throws the following error:

... logging to /home/afroza/.ros/log/0ea56abe-55ce-11e4-ae6d-0023aee70050/roslaunch-afroza-XPS-730X-4531.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

Traceback (most recent call last):
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/__init__.py", line 279, in main
    p.start()
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/parent.py", line 257, in start
    self._start_infrastructure()
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/parent.py", line 206, in _start_infrastructure
    self._load_config()
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/parent.py", line 121, in _load_config
    self.config = roslaunch.config.load_config_default(self.roslaunch_files, self.port, verbose=self.verbose)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/config.py", line 428, in load_config_default
    loader.load(f, config, verbose=verbose)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 698, in load
    self._load_launch(launch, ros_config, is_core=core, filename=filename, argv=argv, verbose=verbose)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 670, in _load_launch
    self._recurse_load(ros_config, launch.childNodes, self.root_context, None, is_core, verbose)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 605, in _recurse_load
    n = self._node_tag(tag, context.child(''), ros_config, default_machine, verbose=verbose)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 95, in call
    return f(*args, **kwds)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 351, in _node_tag
    self.opt_attrs(tag, context, ('machine', 'args', 'output', 'respawn', 'cwd', 'launch-prefix', 'required'))
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 178, in opt_attrs
    return [self.resolve_args(tag_value(tag,a), context) for a in attrs]
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 159, in resolve_args
    return substitution_args.resolve_args(args, context=context.resolve_dict, resolve_anon=self.resolve_anon)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/substitution_args.py", line 299, in resolve_args
    resolved = _resolve_args(resolved, context, resolve_anon, commands)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/substitution_args.py", line 312, in _resolve_args
    resolved = commands[command](resolved, a, args, context)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/substitution_args.py", line 137, in _find
    return _find_executable(resolve_without_path, a, [args[0], path], context)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roslaunch/substitution_args.py", line 173, in _find_executable
    full_path = _get_executable_path(rp.get_path(args[0]), path)
  File "/usr/lib/pymodules/python2.7/rospkg/rospack.py", line 190, in get_path
    raise ResourceNotFound(name, ros_paths=self._ros_paths)
ResourceNotFound: gazebo_worlds
ROS path [0]=/opt/ros/hydro/share/ros
ROS path [1]=/home/afroza/catkin_ws/src
ROS path [2]=/opt/ros/hydro/share
ROS path [3]=/opt/ros/hydro/stacks

Can anybody tell me how can I fix this? your help is highly appreciated.
Thank you
EDIT update
I changed the launch file as:
<launch> <node name="gazebo" pkg="gazebo" type="gazebo" args="-u $(find turtlebot_gazebo)/worlds/playground.world"/> <node name="gazebo_gui" pkg="gazebo" type="gui"/> <include file="$(find turtlebot_teleop)/launch/keyboard_teleop.launch"/> </launch>

Now I am getting the error:
ERROR: cannot launch node of type [gazebo/gazebo]: gazebo ROS path [0]=/opt/ros/hydro/share/ros ROS path [1]=/home/afroza/catkin_ws/src ROS path [2]=/opt/ros/hydro/share ROS path [3]=/opt/ros/hydro/stacks ERROR: cannot launch node of type [gazebo/gui]: gazebo ROS path [0]=/opt/ros/hydro/share/ros ROS path [1]=/home/afroza/catkin_ws/src ROS path [2]=/opt/ros/hydro/share ROS path [3]=/opt/ros/hydro/stacks

Can anybody please help me. Thank you

Originally posted by Ahyan on ROS Answers with karma: 1 on 2014-10-17
Post score: 0

Original comments
Comment by bvbdort on 2014-10-17:
In your catkin_ws folder and do source devel/setup.sh
Comment by Ahyan on 2014-10-17:
I did as:
afroza@afroza-XPS-730X:~/catkin_ws$ source devel/setup.bash
afroza@afroza-XPS-730X:~/catkin_ws$ roslaunch ros_liv_turtlebot office_turtlebot.launch
However, the error still exists.
Thank you
Comment by bvbdort on 2014-10-20:
I moved your edit from answer to question.
Comment by bvbdort on 2014-10-20:
try installing gazebo_ros_pkgs package again with sudo apt-get install ros-hydro-gazebo-ros-pkgs ros-hydro-gazebo-ros-control
Comment by Ahyan on 2014-10-20:
Thank you for your help. I reinstalled gazebo_ros_pkgs package as you suggested. But still I am getting the error.
Comment by bvbdort on 2014-10-21:
use this command to know the path of world files location locate *.world and let me know the path of simple_office.world also the presentation link you shared is based on fuerte.
Comment by Ahyan on 2014-10-31:
Thank you again for your help. I have started again to fix the problem.
As you may notice I am using playground.world not the simple_office.world. I have changed the launch file accordingly. The location of the file is: ~/opt/ros/hydro/share/turtlebot_gazebo/worlds/playground.world. Thank you

A:

gazebo_worlds is Merged to gazebo_ros package , check this document which is helpful.

Originally posted by bvbdort with karma: 3034 on 2014-10-21
This answer was ACCEPTED on the original site
Post score: 0

